### PR TITLE
Fix retry-publish

### PR DESCRIPTION
**Pull request by dciangot**
That view was probably recently changed, returning 2 keys. Fixed and adapted the request accordingly.

Fixing  #4483

### CODE REVIEW

File: bin/retry-publish [hunk 1]
```diff
@@ -76,13 +75,13 @@ Examples:
...
         database = server.connectDatabase(config.AsyncTransfer.files_database)
-    except Exception, ex:
+    except Exception as ex:
         msg = "Error while connecting to %s couchDB." % config.AsyncTransfer.files_database
         print msg
         sys.exit(4)
     try:
-        docs_pubFailed = database.loadView('DBSPublisher','PublicationFailedByWorkflow',{'reduce':False,'key':taskname})['rows']
-    except Exception, ex:
+        docs_pubFailed = database.loadView('DBSPublisher','PublicationFailedByWorkflow',{'reduce':False,'startkey':[taskname,'']})['rows']
```
Comment (HassenRiahi) [inline bin/retry-publish]: It does not seem to work. You need to use the endkey as well: 

{'reduce':False,'startkey':[task], 'endkey':[task, {}]}

**Comment by HassenRiahi** [comment]
Pls @dciangot could you have a look? this patch does not seem to work.

**Comment by dciangot** [comment]
Are you sure? I just tried again and it works just fine for me (also checked the publication state of the docs and they are changed correctly).

**Comment by HassenRiahi** [comment]
> Are you sure? I just tried again and it works just fine for me (also checked the publication state of the docs and they are changed correctly). 

Do you get only the records that match 'task' ?

**Comment by dciangot** [comment]
oh ok, it worked correctly cause of  `if doc['key'][0] != taskname`  control. 
But the view returns all the tasks, so yeah it has to be changed indeed.

**Pull request merged**